### PR TITLE
feat(core)!: Implement floating UI for tooltip positioning and styling

### PR DESCRIPTION
**Pull request by alejo0o**
This PR adds [Floating UI](https://floating-ui.com/docs/flip) in order to handle tooltip position which was having issues on IPad screens. This implementation delegates alignment and positioning to floating UI this way the user can configure floating library for each tourstep or provide a global configuration in order to show the tooltip in different positions, provide offsets, add floating middlewares,etc.

### CODE REVIEW

File: README.md
Comment (JoseLion) [inline README.md:14]: Thanks for the corrections. Maybe we should also explain that the library uses `floting-ui/react-native` and re-exports all their middlewares so they can be used on the configuration, wdyt? 🙂

File: package/src/lib/SpotlightTour.context.ts
Comment (JoseLion) [inline package/src/lib/SpotlightTour.context.ts:81]: Missing docs on the interface and each property. Here you can also document what's the default value of each property (if any)

File: package/src/lib/SpotlightTour.context.ts
Comment (JoseLion) [inline package/src/lib/SpotlightTour.context.ts:95]: Maybe we should add a `@default` to the docs here so people know what they have if they omit this prop 🙂

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:39]: And here too `@default` in docs 🙂

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:103]: Nit: Maybe change this to `FLOATING_PROPS` or `DEFAULT_FLOATING_PROPS`? I'd try to keep it simple, but your call 😬

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:93]: There's a big reformat we don't need in this component 😓 let's revert it also to make the changes easy to go through.

File: package/src/lib/components/tour-overlay/shapes/CircleShape.component.tsx
Comment (JoseLion) [inline package/src/lib/components/tour-overlay/shapes/CircleShape.component.tsx:16]: Same on this one 😭 unnecessary style reformat. Let's revert it too

**Review by JoseLion — COMMENTED** [review]
Looking good so far! There're some code reformats that are making it hard to review precisely what has changed. I left a few suggestions as well 🙂

File: example/src/App.tsx
Comment (well1791) [inline example/src/App.tsx:9]: @alejo0o After looking at the example, my line of thought here is that `react-native-spotlight-tour` is exporting components but maybe something like `react-native-spotlight-tour/floating-middlewares` should import those other operations `flip, offset, shift` and so on, let me know what do you think

File: package/src/lib/SpotlightTour.provider.tsx
Comment (alejo0o) [reply package/src/lib/SpotlightTour.provider.tsx:103]: Sure i like ```DEFAULT_FLOATING_PROPS ```

File: example/src/App.tsx
Comment (alejo0o) [reply example/src/App.tsx:9]: Hey @well1791 we talked this with @JoseLion and we would rather have those dependencies under spotlight tour since we avoid creating to much import statements in one file

File: package/src/lib/SpotlightTour.provider.tsx
Comment (alejo0o) [reply package/src/lib/SpotlightTour.provider.tsx:93]: already fixed 👍

File: README.md
Comment (JoseLion) [inline README.md:16]:
```suggestion
positioning, it re-exports all floating-ui middlewares to be configured in the tour.
```

File: package/src/lib/SpotlightTour.context.ts
Comment (JoseLion) [inline package/src/lib/SpotlightTour.context.ts:185]: We can simplify this name. What do you think about keeping it as `floatingProps` for consistency with the type? 🙂

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:41]: This is great! Can you also add this info somewhere in the readme? I think that will be really useful for users 🙂

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:187]: I think we can leave the default value here:
```suggestion
    return step ?? { floatingProps, render: () => <></> };
```

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:117]: We don't need to rename this prop (maybe it's just a leftover of the refactor):
```suggestion
    floatingProps = DEFAULT_FLOATING_PROPS,
```

File: package/src/lib/SpotlightTour.provider.tsx
Comment (JoseLion) [inline package/src/lib/SpotlightTour.provider.tsx:223]:
```suggestion
      {isChildFunction(children)
        ? <SpotlightTourContext.Consumer>{children}</SpotlightTourContext.Consumer>
        : <>{children}</>
      }
```

**Review by JoseLion — APPROVED** [review]
This is looking great! I left a few suggestions and comments. Please address them before merging. Thank you @alejo0o 🎉

File: example/src/App.tsx
Comment (JoseLion) [reply example/src/App.tsx:9]: @well1791 another option would be to group the middleware exports in an object and then only export that object, like some sort of enum:
```ts
import { flip, offset, shift } from "@floating-ui/react-native";

export const Middleware = {
  flip,
  offset,
  shift,
};

// ...later on userland

import { Middleware } from "@stackbuilders/react-native-spotlight-tour";

floatingProps:{
  middleware: [Middleware.offset(0), Middleware.shift(), Middleware.flip()],
  placement: "right",
}
```

It's more verbose upon usage IMO, but what do you think? 🤔

Comment (well1791) [reply example/src/App.tsx:9]: @alejo0o the decision makes total sense, also @JoseLion moving it into its own object looks great! Agreed on it being verbose but still, this small improvement makes a point on usability. Think of this (although not realistic enough), there may be a case where a dev want to build a lib based on this and may want to do something like this.

```tsx
import { Middleware } from "@stackbuilders/react-native-spotlight-tour";

// ...
type RandomObject = {
  validMiddleware: keyof typeof Middleware

```

If I were the one needed that, I would appreciate having this in advance ^^

File: package/src/lib/SpotlightTour.context.ts
Comment (alejo0o) [reply package/src/lib/SpotlightTour.context.ts:185]: I dont think we can use floatingProps in the spotlight tour context because the component already receives a floatingProps object configuration this means we will have 2 different variables under the same scope with same name

Comment (alejo0o) [reply package/src/lib/SpotlightTour.context.ts:185]: I dont think we can use floatingProps in the spotlight tour context because the component already receives a floatingProps object configuration this means we will have 2 different variables under the same scope with same name

File: package/src/lib/SpotlightTour.provider.tsx
Comment (alejo0o) [reply package/src/lib/SpotlightTour.provider.tsx:41]: I added this information in the readme and updated the examples since they were using the old align to and positioning

File: package/src/lib/SpotlightTour.provider.tsx
Comment (alejo0o) [reply package/src/lib/SpotlightTour.provider.tsx:117]: This is the object i meant in the previous comment, we have this object coming from the provider and we cant use the same name for the context

Comment (alejo0o) [reply package/src/lib/SpotlightTour.provider.tsx:117]: we can change this one into floatingProps with no problem but the ```floatingUiConfigurations``` should have a different name than ```floatingProps```

File: example/src/App.tsx
Comment (JoseLion) [reply example/src/App.tsx:9]: I agree! We can provide both so people can choose the best way for them. I think we can tackle this on a separate PR thought. Thanks for your inputs, it's always helpful to hear different perspectives 🙂

**Review by JoseLion — APPROVED** [review]
Looks good, awesome change! Thanks for your help @alejo0o 🎉

**Pull request merged**
